electrum/transaction.py

```diff
@@ -1844,24 +1854,42 @@ def get_fee(self) -> Optional[int]:
         except MissingTxInputAmount:
             return None
 
+    def is_flag_valid(self, sighash) -> bool:
```

Comment:
This should be a method of `Sighash`, with prototype/signature e.g.: 
`def is_valid(self) -> bool:`

Reply:
I did not understand how to implement it as a method with prototype/signature.  Right now, I have implemented it as a `classmethod`

Reply:
By prototype/signature, I meant what was on the following line; the function name, input types, output types, etc.
See https://en.wikipedia.org/wiki/Function_prototype

So I would like something like this:
```python
class Sighash(IntEnum):
    ALL = 1
    NONE = 2
    SINGLE = 3
    ANYONECANPAY = 0x80

    def is_valid(cls, sighash) -> bool:
        ...
```
That is, `is_valid` does not need to be a classmethod; and instead of taking `sighash` as a param, just use `self`.

Reply:
Ah, nevermind, this does not actually work with `Sighash(IntEnum)`, only with `Sighash(IntFlag)`.
(As, with IntEnum, `type(Sighash.ALL|Sighash.ANYONECANPAY) == int`).
So let's leave it as-is, as
```python
    @classmethod
    def is_valid(cls, sighash) -> bool:
```